src/aof.c

```diff
@@ -40,191 +40,603 @@
+/* Write the formatted manifest string to disk. */
+int writeAofManifestFile(sds buf) {
+    int ret = C_OK;
+    ssize_t nwritten;
+    int len;
 
+    sds amname = getAofManifestFileName();
+    sds temp_amname = getTempAofManifestFileName();
+    int fd = open(temp_amname,O_WRONLY|O_TRUNC|O_CREAT,0644);
+    if (fd == -1) {
+        serverLog(LL_WARNING,
+                "Can't open the aof manifest file %s: %s",
+                 temp_amname, strerror(errno));
+        ret = C_ERR;
+        goto cleanup;
+    }
+    
+    len = sdslen(buf);
     while(len) {
-        /* If we already got at least an allocated block, try appending
-         * at least some piece into it. */
-        if (block) {
-            unsigned long thislen = (block->free < len) ? block->free : len;
-            if (thislen) {  /* The current block is not already full. */
-                memcpy(block->buf+block->used, s, thislen);
-                block->used += thislen;
-                block->free -= thislen;
-                s += thislen;
-                len -= thislen;
-            }
-        }
+        nwritten = write(fd, buf, len);
 
-        if (len) { /* First block to allocate, or need another block. */
-            int numblocks;
-            size_t usable_size;
-
-            block = zmalloc_usable(sizeof(*block), &usable_size);
-            block->free = usable_size-offsetof(aofrwblock,buf);
-            block->used = 0;
-            block->pos = 0;
-            listAddNodeTail(server.aof_rewrite_buf_blocks,block);
-
-            /* Log every time we cross more 10 or 100 blocks, respectively
-             * as a notice or warning. */
-            numblocks = listLength(server.aof_rewrite_buf_blocks);
-            if (((numblocks+1) % 10) == 0) {
-                int level = ((numblocks+1) % 100) == 0 ? LL_WARNING :
-                                                         LL_NOTICE;
-                serverLog(level,"Background AOF buffer size: %lu MB",
-                    aofRewriteBufferSize()/(1024*1024));
-            }
+        if (nwritten < 0) {
+            if (errno == EINTR) continue;
+            serverLog(LL_WARNING,
+                "Error trying to write the temporary AOF manifest file %s: %s",
+                temp_amname,
+                strerror(errno));
+            ret = C_ERR;
+            goto cleanup;
         }
+
+        len -= nwritten;
+        buf += nwritten;
     }
 
-    /* Install a file event to send data to the rewrite child if there is
-     * not one already. */
-    if (!server.aof_stop_sending_diff &&
-        aeGetFileEvents(server.el,server.aof_pipe_write_data_to_child) == 0)
-    {
-        aeCreateFileEvent(server.el, server.aof_pipe_write_data_to_child,
-            AE_WRITABLE, aofChildWriteDiffData, NULL);
+    if (redis_fsync(fd) == -1) {
+        serverLog(LL_WARNING,"Fail to fsync the temp AOF file %s: %s.", 
+                              temp_amname,
+                               strerror(errno));
```

Comment:
styling. i suggest to indent by 4 rather than align to some position (certainly no the offset of the second argument, but also not the first).
this way, if the log function we call (in this case serverLog) is renamed, we don't need to re-align the arguments.
i also don't see why in this case we wanna designate a separate line per argument (certainly not if we didn't for LL_WARNING)
Feel free to apply to others calls (the reason i commented on this one is because it attracted my eye due to an excessive space on the last line)
```suggestion
        serverLog(LL_WARNING,"Fail to fsync the temp AOF file %s: %s.", 
            temp_amname, strerror(errno));
```